src/qiskit_qasm3_import/converter.py

```diff
@@ -25,114 +23,18 @@
-    "crx": (_std.CRXGate, 1, 2),
-    "cry": (_std.CRYGate, 1, 2),
-    "crz": (_std.CRZGate, 1, 2),
-    "ch": (_std.CHGate, 0, 2),
-    "swap": (_std.SwapGate, 0, 2),
-    "ccx": (_std.CCXGate, 0, 3),
-    "cswap": (_std.CSwapGate, 0, 3),
-    "cu": (_std.CUGate, 4, 2),
-    "CX": (_std.CXGate, 0, 2),
-    "phase": (_std.PhaseGate, 1, 1),
-    "cphase": (_std.CPhaseGate, 1, 2),
-    "id": (
-        lambda: _std.UGate(0, 0, 0),  # Qiskit's "id" gate does not strictly have equal semantics.
-        0,
-        1,
-    ),
-    "u1": (_std.U1Gate, 1, 1),
-    "u2": (_std.U2Gate, 2, 1),
-    "u3": (_std.U3Gate, 3, 1),
-}
-
-_BUILTINS = {
-    "U": Symbol(
-        "U",
-        _std.UGate,
-        types.Gate([types.Angle() for _ in [None] * 3], [types.Qubit()]),
-        Scope.BUILTIN,
-    ),
-    "pi": Symbol("pi", math.pi, types.Float(const=True), Scope.BUILTIN),
-    "π": Symbol("π", math.pi, types.Float(const=True), Scope.BUILTIN),
-    "tau": Symbol("tau", math.tau, types.Float(const=True), Scope.BUILTIN),
-    "τ": Symbol("τ", math.tau, types.Float(const=True), Scope.BUILTIN),
-    "euler": Symbol("euler", math.e, types.Float(const=True), Scope.BUILTIN),
-    "ℇ": Symbol("ℇ", math.e, types.Float(const=True), Scope.BUILTIN),
-}
-
-
-class State:
-    __slots__ = ("scope", "source", "circuit", "symbol_table", "_unique")
-
-    def __init__(self, scope: Scope, source: Optional[str] = None):
-        self.scope = scope
-        self.source = source
-        self.circuit = QuantumCircuit()
-        self.symbol_table = _BUILTINS.copy()
-        self._unique = (f"_{x}" for x in itertools.count())
-
-    def gate_scope(self):
-        """Get a new state for entry to a "gate" scope."""
-        # We use the entire source, because at the moment, that's what all the error messages
-        # expect; the nodes have references to the complete source in their spans.
-        out = State(Scope.GATE, self.source)
-        for name, item in self.symbol_table.items():
-            if (item.scope is Scope.BUILTIN) or (
-                item.scope is Scope.GLOBAL
-                and (
-                    isinstance(item.type, types.Gate)
-                    or isinstance(
-                        item.type, (types.Int, types.Uint, types.Float, types.Angle, types.Duration)
-                    )
-                    and item.type.const
-                )
-            ):
-                out.symbol_table[name] = item
-        return out
-
-    def local_scope(self):
-        """Get a new state on entry to a local block scope."""
-        # pylint: disable=protected-access
-        out = State.__new__(State)
-        out.scope = Scope.LOCAL
-        out.source = self.source
-        out.circuit = self.circuit  # No copy; we want to keep modifying this one.
-        out.symbol_table = self.symbol_table.copy()
-        out._unique = self._unique
-        return out
+from .expression import (
+    ValueResolver,
+    resolve_condition,
+)
```

Comment:
What you have looks better if the linter an black accept it. I guess maybe black made the awkward change, and I subsequently edited it or removed something.

Reply:
It's just because of the trailing comma - Black puts them all on separate lines if there is one, and collapses them (if they fit) if there isn't.  Not a big deal.